File: test/warnings/not_handled_warning/cpp/NotHandledWarningTest.cpp
#include <string>
#include <vector>

#include "gtest/gtest.h"

#include "test_utils/ZserioErrorOutput.h"

class NotHandledWarningTest : public ::testing::Test
{
protected:
    NotHandledWarningTest()
    :   zserioWarnings("warnings/not_handled_warning")
    {}

    const test_utils::ZserioErrorOutput zserioWarnings;
};

TEST_F(NotHandledWarningTest, notHandledWhite)
{
    const std::string warning = "not_handled_warning.zs:15:8: "
            "Enumeration value 'WHITE' is not handled in choice 'EnumParamChoice'.";
    ASSERT_TRUE(zserioWarnings.isPresent(warning));
}

TEST_F(NotHandledWarningTest, notHandledRed)
{
    const std::string warning = "not_handled_warning.zs:15:8: "
            "Enumeration value 'RED' is not handled in choice 'EnumParamChoice'.";
    ASSERT_TRUE(zserioWarnings.isPresent(warning));
}

File: test/warnings/not_handled_warning/python/NotHandledWarningTest.py
import unittest

from testutils import getZserioApi, assertWarningsPresent


class NotHandledWarningTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.warnings = {}
        getZserioApi(__file__, "not_handled_warning.zs", expectedWarnings=2, errorOutputDict=cls.warnings)

    def testNotHandledWhite(self):
        assertWarningsPresent(
            self,
            "not_handled_warning.zs",
            [
                "not_handled_warning.zs:15:8: "
                "Enumeration item 'WHITE' is not handled in choice 'EnumParamChoice'."
            ],
        )

    def testNotHandledRed(self):
        assertWarningsPresent(
            self,
            "not_handled_warning.zs",
            [
                "not_handled_warning.zs:15:8: "
                "Enumeration item 'RED' is not handled in choice 'EnumParamChoice'."
            ],
        )
